inference-engine/src/cldnn_engine/ops/convolution.cpp

```diff
@@ -0,0 +1,328 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cldnn_program.h"
+#include "cldnn_common_utils.h"
+
+#include "ngraph/op/convolution.hpp"
+#include "ngraph/op/binary_convolution.hpp"
+#include "ngraph/op/deformable_convolution.hpp"
+#include "ngraph/op/group_conv.hpp"
+#include "ngraph/op/constant.hpp"
+
+#include "api/convolution.hpp"
+#include "api/deconvolution.hpp"
+#include "api/binary_convolution.hpp"
+#include "api/reshape.hpp"
+#include "api/reorder.hpp"
+
+namespace CLDNNPlugin {
+
+
+struct ConvoltuionParameters {
+    cldnn::tensor stride;
+    cldnn::tensor padding;
+    cldnn::tensor dilation;
+    uint32_t groups;
+};
+
+static ConvoltuionParameters GetConvolutionParameters(const ngraph::CoordinateDiff& pads_begin,
+                                                      const ngraph::Strides& dilations,
+                                                      const ngraph::Strides& strides,
+                                                      uint32_t groups) {
+    cldnn::tensor stride, padding, dilation;
+    if (pads_begin.size() != strides.size() || dilations.size() != strides.size())
+        THROW_IE_EXCEPTION << "Strides, Dilations and Pads are supposed to have the same elements count";
+
+    switch (strides.size()) {
+        case 3: {
+            stride = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(strides[2], strides[1], strides[0]));
+            padding = cldnn::tensor(cldnn::batch(0), cldnn::feature(0), cldnn::spatial(-pads_begin[2], -pads_begin[1], -pads_begin[0]));
+            dilation = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(dilations[2], dilations[1], dilations[0]));
+            break;
+        }
+        case 2: {
+            stride = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(strides[1], strides[0], 1));
+            padding = cldnn::tensor(cldnn::batch(0), cldnn::feature(0), cldnn::spatial(-pads_begin[1], -pads_begin[0], 0));
+            dilation = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(dilations[1], dilations[0], 1));
+            break;
+        }
+        case 1: {
+            stride = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(strides[0], 1, 1));
+            padding = cldnn::tensor(cldnn::batch(0), cldnn::feature(0), cldnn::spatial(-pads_begin[0], 0, 0));
+            dilation = cldnn::tensor(cldnn::batch(1), cldnn::feature(1), cldnn::spatial(dilations[0], 1, 1));
+            break;
+        }
+        default: THROW_IE_EXCEPTION << "Unsupported convolve parameters size. Only 1d, 2d, and 3d cases are supported";
+    }
+
+    return {stride, padding, dilation, groups};
+}
+
+void CreateGroupConvolutionOp(Program& p, const std::shared_ptr<ngraph::Node>& node) {
+    auto op = std::dynamic_pointer_cast<ngraph::op::v1::GroupConvolution>(node);
```

Comment:
Such pointer conversion and input number validation is stereotypical piece of code in all **ops** functions.
Can we move this functionality into **REGISTER_FACTORY_IMPL** and **REGISTER_FACTORY** macros?
Besides of getting rid of extra code lines, in future this will also help to faster find and correct type and number of arguments if NGrpah spec changes.

Reply:
Good point. I moved cast to the factory.
As for inputs count validation, I keep it as is for now, since I think we need to refactor this part and extend for more checks. We can move it later in separate PR.